### ISSUE TITLE
First class GPG support

### ISSUE THREAD

**Issue by romanofski** [romanofski/purebred#14]
Signing and encrypting is important. I'm just not clear if it should be in the 1.0 release or not.

**Comment by frasertweedale** [comment]
On Wed, Jul 19, 2017 at 12:36:47AM +0000, Roman Joost wrote:
> Signing and encrypting is important. I'm just not clear if it should be in the 1.0 release or not.
> 
Baseline for sure.  PGP-MIME at least.  (I'm ambivalent on inline
signing/encryption).

I'd like to make it a plugin.  It would be a good target to ensure
an adequate plugin interface for processing messages.

**Comment by frasertweedale** [comment]
I want to remove this from the 1.0 milestone.  It will probably end up as a plugin anyway, although the required hooks and knobs still need to be put into purebred.  But I just don't feel it's something that should hold up a v1.0 release.

**Comment by romanofski** [comment]
Yeah this actually reminds me, that we could make the currently implemented piping action more general to not only work on `WireEntity` but entire `MIMEMessage`s. Having said that, that being able to pipe a message through `gpg2 -d` might not be enough in case the message is not a text format.
All I'm saying is, that my initial thought of supporting this with a more general pipe action won't probably cut it.
I'll move it into a future feature.

**Comment by frasertweedale** [comment]
@romanofski I'd much prefer to work at the MIME AST level, which improves resistance to attacks like https://efail.de/.  I'm fine if we don't even support inline encryption.  But we can decide later :)
